### ISSUE TITLE
Reference :ref:`sphx_glr_auto_examples` goes to SciKit-Learn examples

### ISSUE THREAD

**Issue by riedgar-ms** [sphinx-gallery/sphinx-gallery#734]
We are using `sphinx-gallery` to generate samples for our package. However, I've noticed that in our ReST files, attempts to link to the examples using:
```
:ref:`sphx_glr_auto_examples`
```
result in links to the SciKit-Learn examples. This is undoubtedly due to the inclusion of sklearn in our `intersphinx_mapping` ... but we would like to be able to send readers to our own examples. Is there some way we can override the reference name for ours?

Our docs are built using `sphinx-gallery` 0.7.0, on Linux.

**Comment by larsoner** [comment]
Have you tried being more specific about the name, for example `mylibrary:sphx_glr_auto_examples`?

**Comment by riedgar-ms** [comment]
Thanks - prepending the `:ref:` with `:mylibrary:` did the trick.

**Comment by riedgar-ms** [comment]
How do I do that, exactly (sorry, I realise this is more a ReST question)... if I try
```
:ref:`fairlearn:sphx_glr_auto_examples`
```
I get an 'undefined label' warning (whether or not I capitalise `Fairlearn`).

```
:ref:fairlearn:`sphx_glr_auto_examples`
```
Results in `Unknown interpreted text role "ref:fairlearn` (also with `Fairlearn`), as does:
```
:fairlearn:`sphx_glr_auto_examples`
```

**Comment by larsoner** [comment]
In our `sphinx_gallery/tests/tinybuild/index.rst`, if I add:
```
The complete Sphinx-Gallery project is automatically documented for
every module. Examples :ref:`here <sklearn:sphx_glr_auto_examples>`.
```
and add this to the intersphinx in `conf.py`:
```
    'sklearn': ('https://scikit-learn.org/stable', None),
    'mne': ('https://mne.tools/stable', None),
```
then do `make`, the HTML that pops up has a link to https://scikit-learn.org/stable/auto_examples/index.html#sphx-glr-auto-examples. If I change the text to `here <mne:sphx_glr_auto_examples>`, it links to https://mne.tools/stable/auto_examples/index.html#sphx-glr-auto-examples. So appending the name as `whatever:sphx_glr_auto_examples` should have an effect. And then if I just change the text to:
```
The complete Sphinx-Gallery project is automatically documented for
every module. Examples :ref:`here <sphx_glr_auto_examples>`.
```
it links to my local machine, as it should (`file:///.../sphinx-gallery/sphinx_gallery/tests/tinybuild/_build/html/auto_examples/index.html#sphx-glr-auto-examples`). So the local intersphinx/docs should take precedence over others.

TL;DR: I cannot replicate this issue in our `tinybuild` minimal script.

Perhaps for some reason your module doesn't have intersphinx being generated for it at all? Maybe consider doing a `python -m sphinx.ext.intersphinx _build/html/objects.inv > objects.txt` and look at what's in there. For `tinybuild` there is an entry under `std:label` for `sphx_glr_auto_examples`. Maybe you don't have this for some reason? This would indicate some autosummary/autodoc/intersphinx bug in your docs...

**Comment by riedgar-ms** [comment]
Thanks - running `python -m sphinx.ext.intersphinx _build/html/objects.inv` revealed that I was indeed using the wrong reference. It changed when we put subdirectories into the examples. Now that I've got the right reference out of the inventory, it works.

Thank you for your patience.

**Comment by GaelVaroquaux** [comment]
> revealed that I was indeed using the wrong reference.

That a strange manifestation of such an error.
